core/src/lib.rs

```diff
@@ -40,3 +40,20 @@ pub use self::from_variant::FromVariant;
 pub use quote::ToTokens;
 #[doc(hidden)]
 pub use syn;
+
+#[macro_export]
```

Comment:
Please add a doc comment. Your example from the PR description was great, so I'd suggest including that in the doc comment after the explanation of the macro.